src/Create.jsx

```diff
@@ -329,82 +198,65 @@ const Create = () => {
-        } catch (e) {
-            return false;
-        }
+    const invalidateAddress = (input, msg, setSignal) => {
+        setSignal(false);
+        input.setCustomValidity(msg);
+        setButtonLabel(msg);
+        input.classList.add("invalid");
     };
 
-    const validateAddress = (input) => {
-        let inputValue = input.value.trim();
+    const validateAddress = () => {
         if (reverse()) {
+            const input = addressInputRef;
+            const inputValue = input.value.trim();
             try {
-                // validate btc address
-                const asset_name = asset();
-                const address = getAddress(asset_name);
-                address.toOutputScript(inputValue, getNetwork(asset_name));
+                setOnchainAddress(validateOnchainAddress(inputValue, asset()));
+                setAddressValid(true);
                 input.setCustomValidity("");
                 input.classList.remove("invalid");
-                setAddressValid(true);
-                setOnchainAddress(inputValue);
             } catch (e) {
-                setAddressValid(false);
-                input.setCustomValidity("invalid address");
-                input.classList.add("invalid");
+                invalidateAddress(
+                    input,
+                    "Invalid onchain address",
+                    setAddressValid,
+                );
             }
         } else {
-            inputValue = trimLightningPrefix(inputValue);
-
-            const isInputInvoice = isInvoice(inputValue);
-            if (isLnurl(inputValue) || isInputInvoice) {
-                // set receive/send when invoice differs from the amounts
-                // and the input is an invoice
-                if (isInputInvoice && !checkInvoiceAmount(inputValue)) {
-                    try {
-                        const decoded = decodeInvoice(inputValue);
-                        if (decoded.satoshis === null) {
-                            setInvoiceValid(false);
-                            input.setCustomValidity(
-                                "0 amount invoices are not allowed",
-                            );
-                            input.classList.add("invalid");
-                            return;
-                        }
-                        setReceiveAmount(decoded.satoshis);
-                        setSendAmount(calculateSendAmount(decoded.satoshis));
-                        validateAmount();
-                    } catch (e) {
-                        setInvoiceValid(false);
-                        input.setCustomValidity(e);
-                        input.classList.add("invalid");
-                        return;
-                    }
+            const input = invoiceInputRef;
+            const inputValue = input.value.trim();
+            try {
+                if (isLnurl(inputValue)) {
+                    setButtonLabel(t("fetch_lnurl"));
```

Comment:
I don't like this. The fetching is already abstracted away, there is no need to inform the user about that explicitly